### PR TITLE
CE-3115 Scroll TC modal to top if a preselected type is Unknown

### PR DESCRIPTION
**Pull request by adamkarminski**
/cc: @Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js [hunk 1]
```diff
@@ -163,19 +166,18 @@ function ($, w, mw, loader, nirvana, tracker, throbber, labeling) {
...
+
+		$('#template-classification-' + mw.html.escape(preselectedType)).focus();
+		if (preselectedType === 'unknown') {
+			modalInstance.scroll(0);
+		}
 	}
 
 	function processSave(modalInstance) {
-		var newTemplateType = $('#TemplateClassificationEditForm [name="template-classification-types"]:checked').val(),
-			oldTemplateType = '';
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js:172]: The `track` call in the else block below still uses `oldTemplateType` which is now undefined, so it should be updated there too.

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js [hunk 2]
```diff
@@ -86,11 +88,12 @@ function ($, w, mw, loader, nirvana, tracker, throbber, labeling) {
...
 		// Mark selected type
-		$preselectedType = $classificationForm.find('#template-classification-' + templateType);
+		preselectedType = templateType;
+		$preselectedTypeInput = $classificationForm.find('#template-classification-' + preselectedType);
 
-		if ($preselectedType.length !== 0) {
+		if ($preselectedTypeInput.length !== 0) {
 			$classificationForm.find('input[checked="checked"]').removeAttr('checked');
-			$preselectedType.attr('checked', 'checked');
+			$preselectedTypeInput.attr('checked', 'checked');
```
Comment (Grunny) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js]: `$preselectedTypeInput.prop('checked', true);` is preferred.

Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js]: Yeah, I tried it but strangely - it does not check the selected radio button. Do you think I should investigate further or leave for a future improvement (which may not happen)? :P

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js (hunk 1 above)
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js:172]: Fixed.

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js (hunk 2 above)
Comment (Grunny) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js]: Worked for me from a quick test. Can you send me a link to it not working?

**Comment by azarek** [comment]
Testing done! LGTM

**Pull request merged**